### PR TITLE
refactor: run_umap and unit tests

### PR DESCRIPTION
**Pull request by fangliu117**
Hi @georgezakinih The refactored run_umap function is ready for your review at your convenience. Thank you!
Note: I replaced the previous function name "umap" with "run_umap" to avoid the conflict with umap library name.

### CODE REVIEW

File: tests/test_transformations/test_run_umap.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        self.layer = 'layer1'
+        self.adata.layers[self.layer] = (
+            np.random.rand(10, 10).astype(np.float32)
+        )
+
+    def test_typical_case(self):
+        try:
+            run_umap(self.adata, n_neighbors=5, min_dist=0.1, n_components=2,
+                     random_state=42, layer=self.layer)
+        except Exception as e:
```
Comment (georgezakinih) [inline tests/test_transformations/test_run_umap.py:22]: @fangliu117, why we have these exceptions in the unit tests?

File: tests/test_transformations/test_run_umap.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        umap_dtype = self.adata.obsm["X_umap"].dtype
+        is_floating = np.issubdtype(umap_dtype, np.floating)
+
+        self.assertTrue(is_floating)
+
+    def test_different_parameters(self):
+        try:
+            run_umap(self.adata, n_neighbors=3, min_dist=0.2, n_components=3,
+                     random_state=0, layer=self.layer)
+        except Exception as e:
```
Comment (georgezakinih) [inline tests/test_transformations/test_run_umap.py:61]: @fangliu117, there is no assertion for this test.

**Review by georgezakinih — CHANGES_REQUESTED** [review]

File: tests/test_transformations/test_run_umap.py (hunk 1 above)
Comment (fangliu117) [reply tests/test_transformations/test_run_umap.py:22]: @georgezakinih Thanks for catching this issue. The try-except block in the test_typical_case method was used to capture any exceptions from run_umap and provide a custom failure message using self.fail(), which indicates that the run_umap function raised an exception during its execution. But now I realized it's redundant of catching and re-raising as a test failure here, so I simply this test into:

def test_typical_case(self):
    run_umap(self.adata, n_neighbors=5, min_dist=0.1, n_components=2, 
             random_state=42, layer=self.layer)
    self.assertIn("X_umap", self.adata.obsm)

File: tests/test_transformations/test_run_umap.py (hunk 2 above)
Comment (fangliu117) [reply tests/test_transformations/test_run_umap.py:61]: @georgezakinih Thanks again for raising the error. I revised this test as below:
def test_different_parameters(self):
        run_umap(self.adata, n_neighbors=3, min_dist=0.2, n_components=3, 
                 random_state=0, layer=self.layer)

        # Check that the shape of the UMAP coordinates matches the expected
        umap_shape = self.adata.obsm["X_umap"].shape
        self.assertEqual(umap_shape, (self.adata.n_obs, 3))

**Pull request merged**